ls1intum/Artemis
# [Feature] - Posts anonymization on exercise reset

### Checklist

### Motivation and Context


For our Artemis instance at the University of Innsbruck we have been asked to ensure that Q&A (now Posts) are anonymized after a course has ended. 
Due to this requirement, an anonymization process has been added to the `reset exercise` functionality.

### Description

For the anonymization process, a new anonymous user is created at Application Startup (if not already present) and used to set it as author for Posts and AnswerPosts when the 'reset exercise' is executed.

The content and other attributes of the Posts should not be changed by this. 

### Steps for Testing


1. Log in to Artemis
2. Navigate to a Course's overview
3. Click on an exercise and add Posts / Posts answers
4. Navigate to the Course's management overview and reset the exercise in which the Posts have been added
5. Check if added Posts / Posts answers now have an 'Anonymous' user as author

### Test Coverage




### Screenshots


![Screenshot from 2021-06-07 14-21-53](https://user-images.githubusercontent.com/45761921/121015854-bf672b80-c79b-11eb-8505-31244f634cc9.png)
